Java/src/main/java/com/solace/samples/ManageVPN.java

```diff
@@ -0,0 +1,209 @@
+        }
+    }
+    
+    public void createQueue(String messageVpnName, String queueName ) throws ApiException {
+
+        System.out.format("Creating persistent Queue: %s...\n", queueName);
+           
+        // Create a queue
+        MsgVpnQueue queue = new MsgVpnQueue();
+        queue.setQueueName(queueName);
+        queue.setPermission(PermissionEnum.DELETE);
+        queue.setIngressEnabled(true);
+        queue.setEgressEnabled(true);
+        queue.setAccessType(AccessTypeEnum.NON_EXCLUSIVE);
+        MsgVpnQueueResponse resp = sempApiInstance.createMsgVpnQueue(messageVpnName, queue, null);
+    }
+
+    public void deleteMessageVpn(String messageVPNName) throws ApiException {
+
+        System.out.format("Deleting Message-VPN: %s...\n", messageVPNName);
+
+        // Prerequisite for delete VPN is to remove all queues
+        MsgVpnQueuesResponse queryResp = sempApiInstance.getMsgVpnQueues(messageVPNName, null, null, null, null);
+        List<MsgVpnQueue> queuesList = queryResp.getData();
+        if (queuesList.size() > 0) {
+            System.out.format("Message-VPN contains one or more queues, deleting them first:\n");
+            for (MsgVpnQueue queue: queuesList){
+                deleteQueue(messageVPNName, queue.getQueueName());
+            }
+        }
+    
+        // Delete message-vpn
+        SempMetaOnlyResponse resp = sempApiInstance.deleteMsgVpn(messageVPNName);
+    }
+
+    public void deleteQueue(String messageVpnName, String queueName) throws ApiException {
+
+        System.out.format("Deleting Queue: %s...\n", queueName);
+
+        // Delete queue
+        SempMetaOnlyResponse resp = sempApiInstance.deleteMsgVpnQueue(messageVpnName, queueName);
+    }
+
+    public static void main(String... args) throws Exception {
+
+        // Modify these params as needed
+        final String vmrUser = "admin";
+        final String vmrPassword = "admin";
+        final String vpnUserName = "default";
+        final String vpnUserPassword = "password";
+        final String testQueueName = "testQueue";
+
+        final String usage = "\nUsage: manageVPN [createvpn | deletevpn] <host:port> <vpnname>" +
+                "\nEx: manageVPN createvpn <host:port> <vpnname>" +
+                "\n        Create a message-vpn and add a sample queue: testQueue" +
+                "\n    manageVPN deletevpn <host:port> <vpnname>" +
+                "\n        Delete the message-vpn";
+        
+        // Check command line arguments
+        if (args.length < 3) {
+            System.out.println(usage);
+            System.exit(-1);
+        }
+        
+        String command = args[0];
+        String vmrBasePath = "http://" + args[1] + "/SEMP/v2/config";
+        String messageVpnName = args[2];
+
+        ManageVPN app = new ManageVPN();
+        
+        app.initialize(vmrBasePath, vmrUser, vmrPassword);
+        try {
+            switch (command)  {
+                case "createvpn":
```

Comment:
Could we ignore case in this comparison. For example: http://stackoverflow.com/questions/19628157/switch-ignore-case-in-java-7